tools/clang/lib/AST/ASTContextHLSL.cpp

```diff
@@ -440,11 +440,10 @@ static void AddRecordAccessMethod(clang::ASTContext &Ctx,
     IndexParam->setDefaultArg(ConstantZero);
   }
 
-  StringRef OpcodeGroup = GetHLOpcodeGroupName(HLOpcodeGroup::HLIntrinsic);
-  unsigned Opcode =
-      static_cast<unsigned>(IntrinsicOp::IOP_ExtractRecordStructFromArray);
+  StringRef OpcodeGroup = GetHLOpcodeGroupName(HLOpcodeGroup::HLSubscript);
+  unsigned Opcode = static_cast<unsigned>(HLSubscriptOpcode::DefaultSubscript);
   MethodDecl->addAttr(HLSLIntrinsicAttr::CreateImplicit(
-      Ctx, OpcodeGroup, "ExtractRecordStructFromArray", Opcode));
+      Ctx, OpcodeGroup, "DefaultSubscript", Opcode));
```

Comment:
I think `HLSubscript` group always uses empty string here.

Reply:
How was this resolved?  I meant that we should pass `""` instead of `"DefaultSubscript"`.  This will impact the AST tests as well.

Reply:
Ah, I thought you meant OpcodeGroup will always have the value of "" so I might as well pass "".
Regardless, the reason I switched it back is because leaving it an empty string will cause the preprocess destructor to throw an exception, causing some spir-v tests to fail. It's a strange bug that will take a lot of time to fix, but perhaps it should be left to a separate PR.

Reply:
I'm confused how spir-v tests are affected by this as this is workgraph specific code

Reply:
I am also perplexed.  This is already used with empty string elsewhere for the DefaultSubscript case.  Could it be a bad incremental build?

Reply:
I've made the change before, and the testing pipeline failed. I've run Dr. Memory to test for memory leaks and there are a couple, but it could potentially be because the exception was thrown before the code that would free the memory could execute, so I'm not sure. I do know that the phrase "bad heap allocation" shows up when running the test.
Alternatively, would it make sense to change all cases where HLSubscript uses empty string and instead make it a string that references the opcode? Why must it be an empty string?